multistep/commonsteps/http_config.go

```diff
@@ -58,6 +58,8 @@ type HTTPConfig struct {
 	// interface with a non-loopback address. Either `http_bind_address` or
 	// `http_interface` can be specified.
 	HTTPInterface string `mapstructure:"http_interface" undocumented:"true"`
+	// If true the HTTP server will only be bound to an IPv4 interface
+	HTTPOnlyIPv4 bool `mapstructure:"http_only_ipv4"`
```

Comment:
If we're going with this, I would suggest allowing to specify the protocol here instead of adding a flag, this way we can repurpose it for tcp6/unix or any other protocol later if there's a need for that.

Something like:

```go
HTTPNetworkProtocol string `mapstructure:"http_network_protocol"`
```

Which could be specified as any network type that `Listen` accepts (`Prepare` would then be responsible for ensuring the value is valid).

Reply:
Hi @lbajolet-hashicorp,

Sorry, I've been offline. Thanks a lot for looking at these pull-requests. I have pushed a new commit with the proposed changes.